cmd/fleetctl/fleetctl/gitops.go

```diff
@@ -260,27 +260,21 @@ func gitopsCommand() *cli.Command {
 
 				// Check if any used labels are not in the proposed labels list.
 				// If there are, we'll bail out with helpful error messages.
-				unknownLabelsUsed := false
 				for labelUsed := range labelsUsed {
 					if slices.Index(proposedLabelNames, labelUsed) == -1 {
 						if _, ok := storedLabelNames[fleet.LabelTypeBuiltIn][labelUsed]; ok {
-							logf(
-								fmt.Sprintf(
-									"[!] '%s' label is built-in. Only custom labels are supported. If you want to target a specific platform please use 'platform' instead. If not, please create a custom label and try again. \n",
-									labelUsed,
-								),
+							return fmt.Errorf(
+								"[!] '%s' label is built-in. Only custom labels are supported. If you want to target a specific platform please use 'platform' instead. If not, please create a custom label and try again. \n",
+								labelUsed,
 							)
-						} else {
-							for _, labelUser := range labelsUsed[labelUsed] {
-								logf("[!] Unknown label '%s' is referenced by %s '%s'\n", labelUsed, labelUser.Type, labelUser.Name)
-							}
 						}
-						unknownLabelsUsed = true
+
+						for _, labelUser := range labelsUsed[labelUsed] {
+							logf("[!] Unknown label '%s' is referenced by %s '%s'\n", labelUsed, labelUser.Type, labelUser.Name)
+						}
+						return errors.New("Please create the missing labels, or update your settings to not refer to these labels.")
 					}
 				}
-				if unknownLabelsUsed {
-					return errors.New("Please create the missing labels, or update your settings to not refer to these labels.")
-				}
```

Comment:
I think this was mostly fine thew way it was, just maybe move the `unknownLabelsUsed = true` inside the `else` so that we only get the "please create missing labels" message if there are actually missing labels?  If we return early then we don't get the whole list of missing labels, which makes debugging easier.